llvm/lib/Transforms/Instrumentation/RealtimeSanitizer.cpp

```diff
@@ -45,6 +46,26 @@ static void insertCallAtAllFunctionExitPoints(Function &Fn,
         insertCallBeforeInstruction(Fn, I, InsertFnName);
 }
 
+static PreservedAnalyses rtsanPreservedCFGAnalyses() {
+  PreservedAnalyses PA;
+  PA.preserveSet<CFGAnalyses>();
+  return PA;
+}
+
+static void insertNotifyBlockingCallAtFunctionEntryPoint(Function &F) {
+  IRBuilder<> Builder(&F.front().front());
+  Value *NameArg = Builder.CreateGlobalString(demangle(F.getName()));
+
+  FunctionType *FuncType =
+      FunctionType::get(Type::getVoidTy(F.getContext()),
+                        {PointerType::getUnqual(F.getContext())}, false);
+
+  FunctionCallee Func = F.getParent()->getOrInsertFunction(
+      "__rtsan_notify_blocking_call", FuncType);
+
+  Builder.CreateCall(Func, {NameArg});
```

Comment:
Should we just combine this with a better version of `insertCallAtFunctionEntryPoint` that takes an optional args value? What are the pros/cons of that?

I know in my loop code, I will also need to insert a call here, this time with two variables to the input function. We then have two approaches:

1. Write well named helper functions that do that action and that action only (like you have here)
2. Write a general helper that takes in any number of arguments, which you construct outside the function.

Which do you feel is more appropriate?

Reply:
In my prototype branch, you can see what the second one would look like here:

https://github.com/cjappl/llvm-project/pull/6/files#diff-a0897f69407ad91c84a9618026d8264f776688677e6a49e53164b9bf469f89afR28

I'm down for either approach, but I want to hear your thoughts on one over the other

Reply:
Thanks for the suggestion. I was originally intending to keep the diff minimal, but I prefer refactoring for code re-use. I've pushed up a change that I think tidies it up nicely - let me know what you think!